ttnn/cpp/ttnn/reports.hpp

```diff
@@ -125,57 +133,60 @@ struct BufferPageInfo {
-                buffer_page_info.page_size = page_size;
-                buffer_page_info.buffer_type = buffer->buffer_type();
-                buffer_page_infos.push_back(buffer_page_info);
+
+            auto device_id = device->id();
+            auto address = buffer->address();
+
+            uint32_t page_size = buffer->page_size();
+            auto num_pages = buffer->num_pages();
+            auto num_banks = device->num_banks(buffer->buffer_type());
+
+            if (buffer->buffer_layout() == tt::tt_metal::TensorMemoryLayout::INTERLEAVED) {
+                uint32_t bank_id = 0;
+                for (int page_index = 0; page_index < num_pages; page_index++) {
+                    auto page_address = buffer->page_address(bank_id, page_index);
+                    auto core = buffer->logical_core_from_bank_id(bank_id);
+
+                    BufferPageInfo buffer_page_info = {};
+                    buffer_page_info.device_id = device_id;
+                    buffer_page_info.address = address;
+                    buffer_page_info.core_y = core.y;
+                    buffer_page_info.core_x = core.x;
+                    buffer_page_info.bank_id = bank_id;
+                    buffer_page_info.page_index = page_index;
+                    buffer_page_info.page_address = page_address;
+                    buffer_page_info.page_size = page_size;
+                    buffer_page_info.buffer_type = buffer->buffer_type();
+                    buffer_page_infos.push_back(buffer_page_info);
+
+                    bank_id = (bank_id + 1) % num_banks;
+                }
+            } else {
+                const auto &buffer_page_mapping = *buffer->get_buffer_page_mapping();
+                for (int page_index = 0; page_index < num_pages; page_index++) {
+                    auto dev_page_index = buffer_page_mapping.host_page_to_dev_page_mapping_[page_index];
+                    auto core =
+                        buffer_page_mapping.all_cores_[buffer_page_mapping.dev_page_to_core_mapping_[dev_page_index]];
+                    auto bank_id = device->bank_ids_from_logical_core(buffer->buffer_type(), core)[0];
+                    auto page_address = buffer->sharded_page_address(bank_id, dev_page_index);
+
+                    BufferPageInfo buffer_page_info = {};
+                    buffer_page_info.device_id = device_id;
+                    buffer_page_info.address = address;
+                    buffer_page_info.core_y = core.y;
+                    buffer_page_info.core_x = core.x;
+                    buffer_page_info.bank_id = bank_id;
+                    buffer_page_info.page_index = page_index;
+                    buffer_page_info.page_address = page_address;
+                    buffer_page_info.page_size = page_size;
+                    buffer_page_info.buffer_type = buffer->buffer_type();
+                    buffer_page_infos.push_back(buffer_page_info);
+                }
```

Comment:
Can we do something like this instead?
```
uint32_t bank_id = 0;
for (int page_index = 0; page_index < num_pages; page_index++) {
    tt::tt_metal::CoreLocation core;
    uint64_t page_address = 0;

    if (buffer->buffer_layout() == tt::tt_metal::TensorMemoryLayout::INTERLEAVED) {
        page_address = buffer->page_address(bank_id, page_index);
        core = buffer->logical_core_from_bank_id(bank_id);
        bank_id = (bank_id + 1) % num_banks;
    } else {
        const auto &buffer_page_mapping = *buffer->get_buffer_page_mapping();
        auto dev_page_index = buffer_page_mapping.host_page_to_dev_page_mapping_[page_index];
        core = buffer_page_mapping.all_cores_[buffer_page_mapping.dev_page_to_core_mapping_[dev_page_index]];
        bank_id = device->bank_ids_from_logical_core(buffer->buffer_type(), core)[0];
        page_address = buffer->sharded_page_address(bank_id, dev_page_index);
    }

    BufferPageInfo buffer_page_info = {};
    buffer_page_info.device_id = device_id;
    buffer_page_info.address = address;
    buffer_page_info.core_y = core.y;
    buffer_page_info.core_x = core.x;
    buffer_page_info.bank_id = bank_id;
    buffer_page_info.page_index = page_index;
    buffer_page_info.page_address = page_address;
    buffer_page_info.page_size = page_size;
    buffer_page_info.buffer_type = buffer->buffer_type();

    buffer_page_infos.push_back(buffer_page_info);
}
```

Reply:
Sure I can clean it up. My original change here added a top level loop but didn't touch the logic so all these changes were just indentation.